packages/dcos-integration-test/extra/test_meta.py

```diff
@@ -0,0 +1,82 @@
+"""
+Tests for the integration test suite itself.
+"""
+
+import subprocess
+from pathlib import Path
+from typing import Set
+
+import yaml
+
+from get_test_group import patterns_from_group
+
+
+def _tests_from_pattern(ci_pattern: str) -> Set[str]:
+    """
+    From a CI pattern, get all tests ``pytest`` would collect.
+    """
+    tests = set([])  # type: Set[str]
+    args = ['pytest', '--collect-only', ci_pattern, '-q']
+    result = subprocess.run(args=args, stdout=subprocess.PIPE)
+    output = result.stdout
+    for line in output.splitlines():
+        if b'error in' in line:
+            message = (
+                'Error collecting tests for pattern "{ci_pattern}". '
+                'Full output:\n'
+                '{output}'
+            ).format(
+                ci_pattern=ci_pattern,
+                output=output,
+            )
+            raise Exception(message)
+        if (
+            line and
+            b'PytestWarning' not in line and
+            # Some tests are skipped on collection.
+            b'skipped in' not in line and
+            # Some tests are deselected by the ``pytest.ini`` configuration.
+            b' deselected' not in line and
+            not line.startswith(b'no tests ran in')
+        ):
+            tests.add(line.decode())
```

Comment:
This can crash when the text encoding _inferred_ by this process does not match the text encoding that the child process uses when writing text to stdout.

I know I have said this many many times, but the only reliable way is setting the environment variable PYTHONIOENCODING for the child process; and then we _know_ what output encoding to expect. It might coincidentally always work here, so I am OK if my comment is looked at as purely educational.

Reply:
I will make this change 😄 Context: This was copied without change from E2E code which was written quite some time ago.